### PR TITLE
Toggle auto-assigning personas

### PR DESCRIPTION
**Pull request by MjnMixael**
Adds a global toggle for the persona auto-assignment feature. My mod does not have any personas that aren't tied to specific characters or events so this global mod just blanket turns the feature off. Ships without a persona will never send built-in messages.

### CODE REVIEW

**Review by BMagnu — APPROVED** [review]

File: code/mod_table/mod_table.cpp [hunk 1]
```diff
@@ -1136,6 +1137,10 @@ void parse_mod_table(const char *filename)
 			stuff_boolean(&Hotkey_always_hide_hidden_ships);
 		}
 
+		if (optional_string("$Allow automatically assigning personas:")) {
+			stuff_boolean(&Auto_assign_personas);
```
Comment (Goober5000) [inline code/mod_table/mod_table.cpp:1141]: Almost every boolean flag in game_settings or ai_profiles is something that defaults to false and is optionally set to true.  This should be consistent with that, e.g. `$Do not automatically assign personas:`

**Review by Goober5000 — CHANGES_REQUESTED** [review]

File: code/mod_table/mod_table.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/mod_table/mod_table.cpp:1141]: When you go adjust `$AI use framerate independent turning:` and the few ai_profiles flags I found to make this a true statement across the board then I'll bother with this. Till then, don't merge it, I guess. I don't have time for petty OCD stuff.

Comment (Goober5000) [reply code/mod_table/mod_table.cpp:1161]: The fact that a standard has not been met in 100% of cases is not a valid reason to abandon the standard.

Comment (Goober5000) [reply code/mod_table/mod_table.cpp:1183]: That's an interesting way of looking at it.  But the original pattern was not "do something that isn't normally done" vs "stop doing something that's normally done", but rather "does this alter the base game behavior".  That produces an easy way to check flags, both in mods and in the code: if it alters the behavior, it's true; if it doesn't alter it, it's false.

Comment (Goober5000) [reply code/mod_table/mod_table.cpp:1183]: Any other coders feel like weighing in?

Comment (wookieejedi) [reply code/mod_table/mod_table.cpp:1183]: Hmm I can see both, if we are favoring modders though having the logic match the flag would be the most useful for modders IMO.

**Pull request merged**